src/main/java/emissary/output/DropOffPlace.java

```diff
@@ -225,11 +225,11 @@ public List<IBaseDataObject> agentProcessHeavyDuty(final List<IBaseDataObject> p
             final IBaseDataObject tld = payloadList.get(0);
 
 
-            if (outputCompletionPayloadSize && tld.data() != null) {
+            if (outputCompletionPayloadSize && tld.getChannelFactory() != null) {
```

Comment:
> Is it worth creating a `hasChannelFactory` method to avoid the creation of multiple `ChannelFactory` objects when the ibdo is configured with a byte array?

Creating a channel factory is relatively cheap, since it's jus a wrapper.  The only memory cost got is allocation for the objects.  It's much more expensive to go the other way, creating a `byte[]` when the BDO is backed by a channel factory, because you have to allocate and fill the entire array.

Reply:
> Would it be easier to just make BDO.DataState and BDO.getDataState() public?

That sounds awful.  We shouldn't have those guts exposed.